### PR TITLE
mpvScripts.youtube-quality: replace with mpvScripts.quality-menu v3.0.1

### PR DESCRIPTION
**Pull request by lunik1**
###### Description of changes

This replaces [`mpvScripts.youtube-quality`](https://github.com/jgreco/mpv-youtube-quality) with the maintained fork, [`mpvScripts.quality-menu`](https://github.com/christoph-heinrich/mpv-quality-menu)

I am not sure how to update `aliases.nix` such that an error noting the change is thrown when a user attempts to install `mpvScripts.youtube-quality`. Adding an entry for `mpvScripts.youtube-quality` causes `error: Alias mpvScripts is still in all-packages.nix` to be thrown when _any_ package under mpvScripts is built. Assistance is appreciated, and I am marking this PR as a draft until that issue is resolved.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by chuangzhu — CHANGES_REQUESTED** [review]

**Comment by slashformotion** [comment]
hopping this get merged soon

**Review by chuangzhu — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -28691,12 +28691,13 @@ with pkgs;
...
     mpv-playlistmanager = callPackage ../applications/video/mpv/scripts/mpv-playlistmanager.nix {};
     mpvacious = callPackage ../applications/video/mpv/scripts/mpvacious.nix {};
+    quality-menu = callPackage ../applications/video/mpv/scripts/quality-menu.nix { };
     simple-mpv-webui = callPackage ../applications/video/mpv/scripts/simple-mpv-webui.nix {};
     sponsorblock = callPackage ../applications/video/mpv/scripts/sponsorblock.nix {};
     thumbnail = callPackage ../applications/video/mpv/scripts/thumbnail.nix { };
     vr-reversal = callPackage ../applications/video/mpv/scripts/vr-reversal.nix {};
-    youtube-quality = callPackage ../applications/video/mpv/scripts/youtube-quality.nix { };
     cutter = callPackage ../applications/video/mpv/scripts/cutter.nix { };
+    youtube-quality = throw "youtube-quality is no longer maintained, use quality-menu instead"; # added 2022-12-26
```
Comment (winterqt) [inline pkgs/top-level/all-packages.nix:28700]: This should go into `pkgs/top-level/aliases.nix`.

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (lunik1) [reply pkgs/top-level/all-packages.nix:28700]: How can I add this to `aliases.nix` without causing `error: Alias mpvScripts is still in all-packages.nix`?

**Comment by lunik1** [comment]
In the meantime, I have added quality-menu to [my NUR](https://github.com/lunik1/nur-packagesl).

**Comment by slashformotion** [comment]
> In the meantime, I have added quality-menu to [my NUR](https://github.com/lunik1/nur-packagesl).

thanks for the package

**Comment by Kranzes** [comment]
Done in https://github.com/NixOS/nixpkgs/pull/243354

**Pull request closed without merging**